### PR TITLE
Enhancements and server role alignement 

### PR DESCRIPTION
**Pull request by ulvida**
[Sister backuppc_server role](https://github.com/UdelaRInterior/ansible-backuppc) has evolvedfor a better management of all BackupPC parameters, particularly the description of shares backed up and inclusion/exclusion options. 

Morover, it seemed more logical to adopt a single uniform syntax for name variables in backuppc config.pl file. 

Finally, an idea that emerged documenting v2.1.X version was developed:no more `backuppc_scripts` flag, scripts are used when paths to files are defined. (Changes quite some code in therole, however)

### CODE REVIEW

File: tasks/set_variables.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      backuppc_post_dump_script: scripts/post_dump.sh
+      backuppc_post_dump_template: etc/pgsql/post_dump.sh.j2
+  when:
+  - backuppc_db_server_type == 'pgsql'
+  - backuppc_post_dump_script is not defined
+
+- name: Define the full path of the pre dump script if defined
+  set_fact:
+    backuppc_pre_dump_script_full: >-
+      {{ backuppc_server_home ~ '/' if backuppc_pre_dump_script is not regex('^/.*')
```
Comment (andrespias) [inline tasks/set_variables.yml]: `else '' ` missing

File: tasks/set_variables.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      {% if backuppc_script_pre_sudo_final %}sudo {% endif -%}
+      {{ backuppc_pre_dump_script_full }}
+  when:
+  - backuppc_DumpPreUserCmd is not defined
+  - backuppc_pre_dump_script is defined
+
+- name: SET_FACT | Define the full path of the post dump script if defined
+  set_fact:
+    backuppc_post_dump_script_full: >-
+      {{ backuppc_server_home ~ '/' if backuppc_post_dump_script is not regex('^/.*')
```
Comment (andrespias) [inline tasks/set_variables.yml]: `else '' ` missing

File: tasks/set_variables.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      {{ backuppc_post_dump_script_full }}
+  when:
+  - backuppc_DumpPostUserCmd is not defined
+  - backuppc_post_dump_script is defined
+
+- name: SET_FACT | Define sudoer backuppc client user command 
+  set_fact:
+    backuppc_sudoer: >-
+      Cmnd_Alias BACKUPS = /usr/bin/rsync{{ ', ' ~ backuppc_pre_dump_script_full
+      if backuppc_script_pre_sudo_final is defined and backuppc_script_pre_sudo_final }}{{ ', ' ~  backuppc_post_dump_script_full
```
Comment (andrespias) [inline tasks/set_variables.yml]: `else '' ` missing

File: tasks/set_variables.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      if backuppc_script_post_sudo_final is defined and backuppc_script_post_sudo_final }}
```
Comment (andrespias) [inline tasks/set_variables.yml]: `else '' ` missing

**Review by andrespias — COMMENTED** [review]
Reading Readme.md...

> Behaviour of scripts and db backups are different for mysql and pgsql. Particularly, for MySQL you must add yourself the scripts, por instance pre_dump.sh and post_dump.sh that do the job of dumping databases, while for PostgreSQL these scripts are built form templates (and they dump only a database). 

Can't we make the role build these scripts from templates for Mysql?

**Comment by ulvida** [comment]
> Reading Readme.md...
> 
> > Behaviour of scripts and db backups are different for mysql and pgsql. Particularly, for MySQL you must add yourself the scripts, por instance pre_dump.sh and post_dump.sh that do the job of dumping databases, while for PostgreSQL these scripts are built form templates (and they dump only a database).
> 
> Can't we make the role build these scripts from templates for Mysql?

Certainly the present behaviour to automate databases backups, limited and different for mysql and pgsql, is not very satisfactory. But this was not in the scope of this refactor and enhancement,  which was to complete backuppc config, and clearly separate server and client roles.
Maybe the best could be to let it to a further version, and to have a better design of the feature: one, several or all databases, create user for backup or given by playbook, optional template of scripts, ...

File: tasks/set_variables.yml (hunk 2 above)
Comment (ulvida) [reply tasks/set_variables.yml]: is this compulsory? I remember I had once the complain, but just once.

Comment (ulvida) [reply tasks/set_variables.yml]: You're right. [The else is optional](https://jinja.palletsprojects.com/en/2.11.x/templates/#if-expression), but if you fall in this implicit case the result is "undefined" and not the empty string we need, so her there is an error.

File: tasks/set_variables.yml (hunk 1 above)
Comment (ulvida) [reply tasks/set_variables.yml]: done.

File: tasks/set_variables.yml (hunk 2 above)
Comment (ulvida) [reply tasks/set_variables.yml]: all else '' done.

**Pull request merged**